osquery/tables/applications/browser_utils.cpp

```diff
@@ -47,12 +46,43 @@ void genExtension(const std::string& uid,
     return;
   }
 
+  pt::ptree messagetree;
+  // Find out if there are localized values for fields
+  if (tree.get<std::string>("default_locale", "") != "") {
+    // Read the localized variables into a second ptree
+    std::string messages_json;
+    std::string localized_path = path + "/_locales/" +
+                                 tree.get<std::string>("default_locale") +
+                                 "/messages.json";
+    if (!forensicReadFile(localized_path, messages_json).ok()) {
+      VLOG(1) << "Could not read file: " << localized_path;
+      return;
+    }
+
+    try {
+      std::stringstream messages_stream;
+      messages_stream << messages_json;
+      pt::read_json(messages_stream, messagetree);
+    } catch (const pt::json_parser::json_parser_error& /* e */) {
+      VLOG(1) << "Could not parse JSON from: " << localized_path;
+      return;
+    }
+  }
+
+  std::string localized_prefix = "__MSG_";
   Row r;
   r["uid"] = uid;
   // Most of the keys are in the top-level JSON dictionary.
   for (const auto& it : kExtensionKeys) {
-    r[it.second] = tree.get<std::string>(it.first, "");
-
+    std::string key = tree.get<std::string>(it.first, "");
+    // If the value is an i18n reference, grab referenced value
+    if (key.compare(0, localized_prefix.length(), localized_prefix) == 0) {
+      r[it.second] = messagetree.get<std::string>(
+          key.substr(6, key.length() - 8) + ".message",
```

Comment:
Here it seems you know the key is at least `6` characters long, if it is 6 or 7, then `- 8` may be an issue. Maybe check that the length is also at least `8 + 1`?